test/test_datasource.py

```diff
@@ -105,13 +106,31 @@ def test_update_copy_fields(self):
             m.put(self.baseurl + '/9dbd2263-16b5-46e1-9c43-a76bb8ab65fb', text=response_xml)
             single_datasource = TSC.DatasourceItem('test', '1d0304cd-3796-429f-b815-7258370b9b74')
             single_datasource._id = '9dbd2263-16b5-46e1-9c43-a76bb8ab65fb'
-            single_datasource._tags = ['a', 'b', 'c']
             single_datasource._project_name = 'Tester'
             updated_datasource = self.server.datasources.update(single_datasource)
 
         self.assertEqual(single_datasource.tags, updated_datasource.tags)
         self.assertEqual(single_datasource._project_name, updated_datasource._project_name)
 
+    def test_update_tags(self):
+        with open(ADD_TAGS_XML, 'rb') as f:
+            add_tags_xml = f.read().decode('utf-8')
+        with open(UPDATE_XML, 'rb') as f:
+            update_xml = f.read().decode('utf-8')
+        with requests_mock.mock() as m:
+            m.put(self.baseurl + '/9dbd2263-16b5-46e1-9c43-a76bb8ab65fb/tags', text=add_tags_xml)
```

Comment:
Just cause there are so many endpoints on this test, can you comment what the endpoints are for?

eg
```
# Updating tags
m.put(self.baseurl + '/9dbd2263-16b5-46e1-9c43-a76bb8ab65fb', text=update_xml)
```